server/plugin.go

```diff
@@ -152,6 +152,13 @@ func (p *Plugin) handleTelemetry(w http.ResponseWriter, r *http.Request) {
 		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to decode JSON", err)
 		return
 	}
+	r.Body.Close()
+
+	if telemetryRequest == nil {
```

Comment:
Why does `telemetryRequest` need to be a pointer is the first place? Can we make it a value?

Reply:
There are two points to distinguish here:
1. Avoid NPE
2. Semantic validation


Using a value does solve 1. Does the rest of the code guard against 2.? Does using a pointer help with 2.?

Reply:
> Does the rest of the code guard against 2.? 

No, In that case the structure will be empty and i don't think the code will be able to handle that. 
> Does using a pointer help with 2.?

Yes, It helps us in checking after unmarshalling that it is nil or not.

Reply:
What happens if `{}` is passed as the payload? Does the code handle that case well?

Reply:
@hanzei yeah i tried that the code is handling this case

Reply:
@hanzei I tested this with every POST API present in plugin like passing the empty body ```{}```. There are checks present in all the APIs apart from 2 which are ```/add``` and ```/telemetry``` which are returning 200 ok in the empty body case. 

Should i add the checks in these 2 APIs as well for the empty body ?

Reply:
> @hanzei I tested this with every POST API present in plugin like passing the empty body `{}`. There are checks present in all the APIs apart from 2 which are `/add` and `/telemetry` which are returning 200 ok in the empty body case.
> 

Q: Do these API endpoints return a 200 and do something useful? Are the operations that these API calls trigger in the DB, valid ones?

> Should i add the checks in these 2 APIs as well for the empty body ?

I would go even further and say that the endpoints should not only check for an empty body but also check if every required field is set.

Reply:
> Q: Do these API endpoints return a 200 and do something useful? Are the operations that these API calls trigger in the DB, valid ones?

@hanzei No I don't think so as in the case of ```/add``` it is just storing the empty values.
> I would go even further and say that the endpoints should not only check for an empty body but also check if every required field is set.

 @hanzei I agree with you we should validate the required set for each and every POST APIs. Will work on it next.

Reply:
Yes, please.